api/catalog/filter.go

```diff
@@ -100,19 +100,21 @@ func RefreshActionHandler(actionName string, action *types.Action, apiContext *t
 
 func extractVersionLinks(apiContext *types.APIContext, resource *types.RawResource) map[string]string {
 	schema := apiContext.Schemas.Schema(&managementschema.Version, client.TemplateVersionType)
-	versionMap := resource.Values["versions"].([]interface{})
 	r := map[string]string{}
-	for _, version := range versionMap {
-		revision := ""
-		if v, ok := version.(map[string]interface{})["revision"].(int64); ok {
-			revision = strconv.FormatInt(v, 10)
-		}
-		version := version.(map[string]interface{})["version"].(string)
-		versionID := fmt.Sprintf("%v-%v", resource.ID, version)
-		if revision != "" {
-			versionID = fmt.Sprintf("%v-%v", resource.ID, revision)
+	versionMap, ok := resource.Values["versions"].([]interface{})
+	if ok {
+		for _, version := range versionMap {
+			revision := ""
+			if v, ok := version.(map[string]interface{})["revision"].(int64); ok {
+				revision = strconv.FormatInt(v, 10)
+			}
+			version := version.(map[string]interface{})["version"].(string)
```

Comment:
Is it safe to assume that the "version" will always be in the map? If not, may be we can continue the loop from this point.